commands/skills/appraise.py

```diff
@@ -0,0 +1,142 @@
+from commands.command import MuxCommand
+from world.economy import value_to_coin
+from world.rulebook import skill_check
+from time import time
+
+class CmdAppraise(MuxCommand):
+    """
+    appraise an item
+
+    Usage:
+      appraise <item>
+      appr <item>
+
+    Indicates the approximate value of an item in your inventory.
+    The higher the Appraise skill, the better the approximation will be.
+    """
+    key = "appraise"
+    aliases = ["appr"]
+    locks = "cmd:all()"
+
+    def func(self):
+        caller = self.caller
+        args = self.args.strip()
+        
+        if not args:
+            caller.msg("Appraise what?")
+            return
+
+        # get the argument of the command        
+        obj = caller.search(args)
+        
+        if not obj:
+            caller.msg("You do not have %s." % args)
+            return
+        
+        if not hasattr(obj, "value"):
+            caller.msg("%s cannot be appraised." % obj.name.capitalize())
+
+        ct = time() # current time; we need this to check if the last appraise
+                    # on the item is recent
+
+            # if they don't exist, create the appraise roll win and lose lists,
+            # which list all items that appraise was attempted on and their
+            # timestamps
+
+        if not hasattr(caller.ndb, "appr_lose") or not caller.ndb.appr_lose:
+            caller.ndb.appr_lose = {}
+        
+        if not hasattr(caller.ndb, "appr_win") or not caller.ndb.appr_win:
+            caller.ndb.appr_win = {}
+
+        # check if the object has failed to be appraised recently. If so,
+        # prohibit further appraisal.
+
+        #caller.msg("initial lose list: " + str(caller.ndb.appr_lose))
+        #caller.msg("initial win list: " + str(caller.ndb.appr_win))
+
+        lose = caller.ndb.appr_lose
+        reroll_time = 3600 # how much time one has to wait before rerolling
+                           # the appraise skill on the same type of item
+
+        if lose.has_key(obj.name):
```

Comment:
Could the win/lose dicts be keyed on `obj.id` instead of `obj.name`? I could see us expanding the functionality of this command to be able to appraise objects other than items in inventory, and we can't necessarily count on every item having a `name` property. Using `id` will give more flexibility going forward.